### DISCUSSION TITLE
Victron - Signal K

### DISCUSSION THREAD

**Discussion by wsturner88** [bareboat-necessities/lysmarine_gen#211] [General]
I am intersted in getting the data from my Victron Shunt into SignalK.  I thought I saw this in the notes..    Can someone point me in the direction of how to set this up?

Thanks for the Package.. I am up and running on a 7". Thinking of upgrading to a 10" screen

**Comment by wsturner88** [comment]
Just a followup - I am running on a monitor 14" and having some issues - When I open OpenCPN it locks up.   I have signalK working great.   I tried increasing GPU Memory to 256 thinking it was the large monitor.   Any other ideas?

Rpi 4 - 4MB running off an SSD
Same issue occurs when booting from a microSD.

**Reply by mgrouch** [reply]
Are you running 64-bit bullseye or 32-bit buster? Bullseye version is new however OpenCPN should work. Locking up is not good. Are you using any plugins. Like o-chart or some other?

**Reply by wsturner88** [reply]
Found the Cause...  If I am disconnected from my NEMA feed (Miniplex) OpenCPN runs find.   about 10 seconds after I connect my network (or NMEA feed) it crashes.   Signal K reports about 80 delta's per second..

**Reply by mgrouch** [reply]
Are you using NMEA 2000? It’s higher data rate. I’ve seen complaints about OpenCPN not being able to handle it from SignalK. Best way to address it is open issue directly against OpenCPN at

https://github.com/OpenCPN/OpenCPN/issues

**Reply by wsturner88** [reply]
mgrouch, I think you are right, I will give that a try tonight.     Yes, I am using NMEA 2000...  I wonder if OpenCPN team knows about this issue..

**Reply by mgrouch** [reply]
They might not know. It’s better to report this issue to them. And at least we need some kind of workaround with current version

**Reply by mgrouch** [reply]
@wsturner88 

I’ve reported it as OpenCPN issue. 

https://github.com/OpenCPN/OpenCPN/issues/2750#issuecomment-1221316364

OpenCPN needs SignalK log of messages which causes lock up in order to fix the issue. 

Would you be able to record such log?

thanks

**Reply by wsturner88** [reply]
I have been looking through signalK and don't see where to capture or log the NMEA data..

I did see SignalK datadog?   Is that an option?

**Reply by mgrouch** [reply]
> I have been looking through signalK and don't see where to capture or log the NMEA data..
> 
> I did see SignalK datadog? Is that an option?

@wsturner88 

To save into sk.log file:

````
wsdump 'ws://localhost:3000/signalk/v1/stream?serverevents=all&subscribe=all&sendMeta=all'  > sk.log


````

from command line

**Comment by mgrouch** [comment]
As far as Victron is concerned. Check SignalK plugins. Some are preinstalled. Homeassistant also offers some Victron integration. You might want to enable homeassistant in BBN OS. There is a section in the docs on how to do it via add-ons scripts

**Reply by mgrouch** [reply]
There are also preinstalled Victron supporting nodes in NodeRed
So there are at least 3 options 
for Victron integration 

SignalK 
NodeRed
Homeassistant

**Reply by wsturner88** [reply]
GOT IT WORKING!   I loaded one of the SignalK plugin's and configured it..  Now the data is available..  Thank you!

**Reply by mgrouch** [reply]
> GOT IT WORKING! I loaded one of the SignalK plugin's and configured it.. Now the data is available.. Thank you!

Which plugin are you using?
Thanks

**Comment by mgrouch** [comment]
Have you had a chance to try Homeassistant Victron integration?  
Have you tried NodeRed with Victron?
It would be nice if you could post here notes how you got it working and what is working. 

thanks

**Reply by mgrouch** [reply]
I've added Victron Venus Web UI into Bullseye branch of BBN Marine OS. It should make it into next image

https://github.com/victronenergy/venus-html5-app

**Reply by wsturner88** [reply]
Wow...  Great..
In regards to rednode and homeassistnant, I have not really done anything with them.   I need to get the issues with SignalK resolved if I can going to continue down this path for my NAV and also dashboard system for the boat.   

I hear Rednode is powerful, just have no idea how to use it..  might be a winter project.

**Reply by mgrouch** [reply]
Starting with version 2022-08-29 and up BBN Marine OS includes Victron Web UI.

Victron web UI in BBN OS is built from:

https://github.com/victronenergy/venus-html5-app

**Comment by wsturner88** [comment]
Cant wait to try...

**Comment by mgrouch** [comment]
See also: https://github.com/bareboat-necessities/lysmarine_gen/discussions/269

**Comment by mgrouch** [comment]
Related issue

https://github.com/bareboat-necessities/lysmarine_gen/issues/277

**Comment by mgrouch** [comment]
Related discussion thread:

https://github.com/bareboat-necessities/lysmarine_gen/discussions/269

**Comment by mgrouch** [comment]
Here is a good article for integration of Victron with Homeassistant on a boat

https://boathackers.com/boat-energy-management-using-victron-and-home-assistant/

**Comment by mgrouch** [comment]
GuiMods for Victron VenusOS
https://github.com/kwindrem/GuiMods
